src/includes/classes/MenuPageOptions.php

```diff
@@ -522,8 +522,14 @@ public function __construct()
             echo '      <p><select name="'.esc_attr(GLOBAL_NS).'[saveOptions][when_logged_in]">'."\n";
             echo '            <option value="0"'.selected($this->plugin->options['when_logged_in'], '0', false).'>'.__('No, do NOT cache; or serve a cache file when a user is logged-in (safest option).', SLUG_TD).'</option>'."\n";
             echo '            <option value="postload"'.selected($this->plugin->options['when_logged_in'], 'postload', false).'>'.__('Yes, and maintain a separate cache for each user (recommended for membership sites).', SLUG_TD).'</option>'."\n";
-            echo '            <option value="1"'.selected($this->plugin->options['when_logged_in'], '1', false).'>'.__('Yes, but DON\'T maintain a separate cache for each user (I know what I\'m doing).', SLUG_TD).'</option>'."\n";
+            if ($this->plugin->options['when_logged_in'] === '1' || get_site_option(GLOBAL_NS.'_when_logged_in_was_1')) {
+                update_site_option(GLOBAL_NS.'_when_logged_in_was_1', '1');
```

Comment:
@jaswsinc This new `_when_logged_in_was_1` option is preparation for "Existing users who have this option enabled" as discussed in https://github.com/websharks/zencache/issues/497#issuecomment-125353814, correct?